re-enable C89 support

Hello!

Thanks for such a high quality library!

Unfortunately, latest has a problem for me:

```
21:38:25 cmp.c: In function 'cmp_write_float':
21:38:25 cmp.c:421: error: 'for' loop initial declaration used outside C99 mode
21:38:25 cmp.c: In function 'cmp_write_double':
21:38:25 cmp.c:439: error: 'for' loop initial declaration used outside C99 mode
```

I sadly have to use C89 for some older plugins and/or I can't add a flag to turn on C99 mode.   Could we pull out the variable from the for loop?  I know it is gross.

Happy to provide a patch as well.

regards,

nickg

--- a/cmp.c
+++ b/cmp.c
@@ -417,8 +417,9 @@ bool cmp_write_float(cmp_ctx_t *ctx, float f) {
   if (!is_bigendian()) {
     char swapped[sizeof(float)];
     char *fbuf = (char *)&f;
+    size_t i;
 
-    for (size_t i = 0; i < sizeof(float); i++)
+    for (i = 0; i < sizeof(float); i++)
       swapped[i] = fbuf[sizeof(float) - i - 1];
 
     return ctx->write(ctx, swapped, sizeof(float));
@@ -435,8 +436,9 @@ bool cmp_write_double(cmp_ctx_t *ctx, double d) {
   if (!is_bigendian()) {
     char swapped[sizeof(double)];
     char *dbuf = (char *)&d;
+    size_t i;
 
-    for (size_t i = 0; i < sizeof(double); i++)
+    for (i = 0; i < sizeof(double); i++)
       swapped[i] = dbuf[sizeof(double) - i - 1];
 
     return ctx->write(ctx, swapped, sizeof(double));

